lib/fpm/package/rpm.rb

```diff
@@ -163,11 +163,15 @@ class FPM::Package::RPM < FPM::Package
   # Replace ? with [?] to make rpm not use globs
   # Replace % with [%] to make rpm not expand macros
   def rpm_fix_name(name)
-    name = "\"#{name}\"" if name[/\s/]
-    name = name.gsub("[", "[\\[]")
-    name = name.gsub("*", "[*]")
-    name = name.gsub("?", "[?]")
-    name = name.gsub("%", "[%]")
+    name = name.gsub(/(\ |\[|\]|\*|\?|\%|\$)/, {
+      ' ' => '?',
```

Comment:
Perhaps we need an option to explicitly define the behavior. But for me, in
mode -t dir, I would expect it to be harmless if it did match both "foo
bar" and "foo.bar". I've made the changes locally, so I'm happy with the
behavior. But it is really broken without the merge request if you have a
file named "foo [*baz%] bar". The quoting of the name stops the glob
expressions from working.

On Thu, 25 Jun 2015 17:28 Jordan Sissel <notifications@github.com> wrote:

> In lib/fpm/package/rpm.rb
> <https://github.com/jordansissel/fpm/pull/946#discussion_r33274683>:
>
> > @@ -163,11 +163,15 @@ class FPM::Package::RPM < FPM::Package
> >    # Replace ? with [?] to make rpm not use globs
> >    # Replace % with [%] to make rpm not expand macros
> >    def rpm_fix_name(name)
> > -    name = "\"#{name}\"" if name[/\s/]
> > -    name = name.gsub("[", "[\\[]")
> > -    name = name.gsub("*", "[*]")
> > -    name = name.gsub("?", "[?]")
> > -    name = name.gsub("%", "[%]")
> > +    name = name.gsub(/(\ |\[|\]|\*|\?|\%|\$)/, {
> > +      ' ' => '?',
>
> This isn't correct, I don't think. Two files: 'foo bar' and 'foo.bar' will
> be matched by the result of this.
>
> I'm not really sure what the right solution here is given rpmbuild's bugs
> with quoted strings and wildcards.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jordansissel/fpm/pull/946/files#r33274683>.
>